rust_dev_preview/cross_service/rest_ses/tests/api/work_item/happy.rs

```diff
@@ -0,0 +1,139 @@
+        ))
+        .send()
+        .await
+        .expect("Request failed");
+
+    eprintln!("{response:?}");
+
+    let status = response.status();
+    let body = response.text().await.expect("response missing a body");
+
+    assert!(
+        status.is_success(),
+        "Response did not return with success: {body}"
+    );
+
+    let work_item: WorkItem =
+        serde_json::from_str(body.as_str()).expect("failed to parse response body");
+
+    let result = rds
+        .execute_statement()
+        .sql("SELECT idwork FROM Work WHERE idwork = :idwork;")
+        .set_parameters(params![("idwork", work_item.idwork())])
+        .send()
+        .await
+        .expect("failed to query rds");
+
+    assert_eq!(
+        result.records().unwrap().len(),
+        1,
+        "should have inserted one record"
+    );
+}
+
+/// Create a single item, and assert we can get it back from the list endpoint.
+#[ignore]
+#[tokio::test]
+async fn get_workitem_returns_200() {
+    let (app, _) = spawn_app().await;
+
+    let name = fake_name();
+    let guide = fake_guide();
+    let description = fake_description();
+
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("{app}/api/items"))
+        .header("content-type", "application/json")
+        .body(format!(
+            r#"{{"name":"{name}","guide":"{guide}","description":"{description}"}}"#
+        ))
+        .send()
+        .await
+        .expect("Request failed");
+
+    let body = response.text().await.expect("response missing a body");
+    let work_item: WorkItem =
+        serde_json::from_str(body.as_str()).expect("failed to parse response body");
+    let id = work_item.idwork();
+
+    eprintln!("\n\nLooking for {id}\n\n");
+
+    let response = client
+        .get(format!("{app}/api/items/{id}",))
+        .send()
+        .await
+        .expect("Failed to look up created item");
+
+    let status = response.status();
+    let body = response.text().await.expect("response missing a body");
+
+    eprintln!("{}", body);
+
+    assert!(
+        status.is_success(),
+        "Response did not return with success: {status} {body}"
+    );
+}
+
+/// Look for an item that shouldn't exist, check for 404.
+#[ignore]
+#[tokio::test]
+async fn get_workitem_returns_404() {
+    let (app, _) = spawn_app().await;
+
+    let id = Uuid::new_v4().to_string();
+    eprintln!("\n\nLooking for {id}\n\n");
+
+    let client = reqwest::Client::new();
+    let response = client
+        .get(format!("{app}/api/items/{id}",))
+        .send()
+        .await
+        .expect("Failed to look up created item");
+
+    let status = response.status();
+    let body = response.text().await.expect("response missing a body");
+
+    eprintln!("{}", body);
+
+    assert_eq!(
+        status,
+        StatusCode::NOT_FOUND,
+        "Response returned as not missing: {status} {body}"
+    );
+}
```

Comment:
What about also testing for getting multiple items, both archived and active?
And for archiving an item?

Reply:
Done!